zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -1357,6 +1375,61 @@ where
         }
         .boxed()
     }
+
+    fn generate(&self, num_blocks: u32) -> BoxFuture<Result<Vec<GetBlockHash>>> {
+        let rpc: GetBlockTemplateRpcImpl<
+            Mempool,
+            State,
+            Tip,
+            BlockVerifierRouter,
+            SyncStatus,
+            AddressBook,
+        > = self.clone();
+        let network = self.network.clone();
+
+        async move {
+            if !network.is_regtest() {
+                return Err(Error {
+                    code: ErrorCode::ServerError(0),
+                    message: "generate is only supported on regtest".to_string(),
```

Comment:
Optional: This method could also work on Testnet if proof of work is disabled (the docs in the trait will need an update as well if this suggestion is applied).

```suggestion
            if !network.disable_pow() {
                return Err(Error {
                    code: ErrorCode::ServerError(0),
                    message: "generate is only supported when PoW is disabled".to_string(),
```